clab/clab.go

```diff
@@ -155,13 +155,33 @@ func (c *CLab) GlobalRuntime() runtime.ContainerRuntime {
 }
 
 func (c *CLab) CreateNodes(ctx context.Context, maxWorkers uint, serialNodes map[string]struct{}) {
+	staticIPNodes := make(map[string]nodes.Node)
+	dynIPNodes := make(map[string]nodes.Node)
 
+	for name, n := range c.Nodes {
+		if n.Config().MgmtIPv4Address != "" || n.Config().MgmtIPv6Address != "" {
+			staticIPNodes[name] = n
+			continue
+		}
+		dynIPNodes[name] = n
+	}
+	log.Debug("scheduling nodes with static IPs first...")
```

Comment:
What if we do this only if static nodes are present? Doing a length check here, as well as keeping the code in createNodes as it is now and just removing the return statement if length is 0?

That way we avoid issuing a debug print here that is not needed if there are now nodes with static IP and won't call a function

Reply:
I fail to see the advantage in that change,
what if all the nodes are with static IPs and the dynamic nodes are 0?
should we do a length for the dynamic IPs as well ?

Reply:
I merely wanted to prevent a log message from printing 

```
log.Debug("scheduling nodes with static IPs first...")
```

if it is known if there are no nodes to schedule. But can be just me

Reply:
I will change it, but I still think it's less optimized.
Scheduling doesn't mean there are node to be scheduled, if there are none, it won't happen.